homeassistant/components/template/switch.py

```diff
@@ -219,3 +297,90 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         if self._template is None:
             self._state = False
             self.async_write_ha_state()
+
+
+class TriggerSwitchEntity(TriggerEntity, SwitchEntity, RestoreEntity):
```

Comment:
Wouldn't it be better to have an `AbstractSwitchTemplate` class from which both `SwitchTemplate` and TriggerSwitchEntity` inherit?

Reply:
It would be a nice idea, however none of the existing template entities do this and there are challenges that arise.  It is something that I want to do, but I want it to be done at a integration level, not domain level.